src/main/java/com/cloudbees/plugins/credentials/CredentialsNameProvider.java

```diff
@@ -90,7 +90,11 @@ Result name(@NonNull Credentials credentials, @NonNull Class<?> clazz) {
         if (nameWith != null) {
             try {
                 CredentialsNameProvider nameProvider = nameWith.value().newInstance();
-                return new Result(nameProvider.getName(credentials), nameWith.priority());
+                String name = nameProvider.getName(credentials);
+                if (!name.isEmpty()) {
```

Comment:
Make it possible to just `return credentials.getDescription()` and if that happens to be the empty string, still behave reasonably (choose a different provider).